lib/screens/home/home_event.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:flutter/cupertino.dart';
+import 'package:meta/meta.dart';
+import 'package:pub_client/pub_client.dart';
+
+@immutable
+abstract class HomeEvent {}
+
+class GetPageOfPackagesEvent extends HomeEvent {
+  final int pageNumber;
+  final SortType sortBy;
+  final FilterType filterBy;
+
+  GetPageOfPackagesEvent({
+    this.pageNumber = 1,
+    this.sortBy,
+    this.filterBy,
+  });
+
+  GetPageOfPackagesEvent copyWith(
+      {int pageNumber, SortType sortBy, FilterType filterBy}) {
+    return GetPageOfPackagesEvent(
+        pageNumber: pageNumber ?? this.pageNumber,
+        sortBy: sortBy ?? this.sortBy,
+        filterBy: filterBy ?? this.filterBy);
+  }
+}
+
+class ChangeFilterTypeEvent extends HomeEvent {
+  final FilterType filterType;
+
+  ChangeFilterTypeEvent(this.filterType);
```

Comment:
for consistency shouldn't this also be a named property? Also shouldn't it be required?

Reply:
It's required. Just not named since it's the Illy parameter. I like to name 2 or more to ease confusion. But if consistency is preferable here, I'm all for it